### PR TITLE
Feat: Defkit Refactor and Clean-Up

### PR DESCRIPTION
**Pull request by roguepikachu**
## Description of your changes

Clean up and harden the `pkg/definition/defkit` package by fixing bugs, removing dead code, consolidating duplicate types, replacing deprecated APIs, and reorganizing `component.go` into focused single-concern files.

### File Reorganization

Split `component.go` (1326 lines) into 4 focused files, zero API changes, pure file moves:
- `template.go`: Template struct and methods
- `patch.go`: PatchResource and patch operations
- `helper_definition.go`: HelperDefinition struct
- `render.go`: Render, RenderAll, and rendering logic

Added 65 new Ginkgo specs covering the reorganized code paths.

### Bug Fixes

- **BuiltinActionBuilder.If()**: Fixed double-append bug where `If()` and `Build()` both appended actions
- **strings.Title**: Replaced deprecated API with manual capitalization
- **appendCombine**: Combined consecutive appends to satisfy linter

### Code Cleanup

- **PatchContainerHelper**: Removed unused struct and methods
- **ParamIsSetCondition / ParamNotSetCondition**: Consolidated duplicate types, added `Not(IsSet("x"))` to `parameter.x == _|_` optimization

### Enhancements

**DeploymentHealth() composable API**: Replaced raw CUE strings with builder API:
- Field grouping with column-aligned defaults
- `WithDefault()` for `_isHealth` pattern
- `WithDisableAnnotation()` for health-check disable
- Added 5 edge-case tests

**Code Documentation**: Clarified internal IR types vs user-facing API.

I have:
- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

## How has this code been tested

- All 593 Ginkgo specs pass (up from 528)
- `make vela-cli` builds successfully
- `vela def apply-module vela-go-definitions --dry-run` output comparison: zero health/status diffs
- `make reviewable` passes clean

## Special notes for your reviewer

Refactoring PR focused on code quality. Main changes: 
1. `component.go` split into 4 focused files with zero API changes
2. DeploymentHealth() builder API replaces raw CUE strings with identical output

### CODE REVIEW

**Comment by roguepikachu** [comment]
The `ParamIsSetCondition` / `ParamNotSetCondition` consolidation eliminates duplicate types. These were duplicates of `IsSetCondition` / `NotCondition` that already existed in `expr.go`. We had two parallel type hierarchies for the same concept, with separate `conditionToCUE()` branches. Now it combines to one code path instead of two.

**Comment by roguepikachu** [comment]
The `workflow_step.go` bug fix is a correctness issue. `If()` + `Build()` was double-appending actions to the template, which would generate incorrect CUE output.

**Comment by roguepikachu** [comment]
HealthBuilder enhancements (`WithDefault`, `WithDisableAnnotation`, `buildGroupedFields`) replace raw CUE string literals with the builder API, which is the whole point of defkit. Without them, `DeploymentHealth()` was just wrapping a hardcoded string instead of using composable builders.

**Comment by roguepikachu** [comment]
The `component.go` split into `template.go`, `patch.go`, `helper_definition.go`, and `render.go` is purely organizational with zero API changes. `component.go` mixed unrelated concerns (template building, patch operations, trait introspection, helper definitions, and test rendering).

File: pkg/definition/defkit/expr.go [hunk 1]
```diff
@@ -186,21 +186,26 @@ func (c *CompareCondition) Right() any { return c.right }
...
+// This is an internal IR type used by cuegen to combine conditions during code generation.
+// For the user-facing API, use Or() which accepts variadic conditions via LogicalExpr.
 type OrCondition struct {
 	baseCondition
 	left  Condition
 	right Condition
 }
 
 // NotCondition represents a logical NOT of a condition.
+// Used both internally (e.g., by ParamNotSet) and as part of the user-facing Not() API.
```
[the remark this answers is not in the document]
Comment (roguepikachu) [reply pkg/definition/defkit/expr.go:208]: This doesn't make much sense, it only adds what it returns, that's pretty intuitive already
